src/Data/Array/Mutable/Linear.hs

```diff
@@ -223,7 +227,7 @@ instance Consumable (Array a) where
   consume (Array arr) = arr `Unlifted.lseq` ()
 
 instance Data.Functor Array where
-  fmap f (Array arr) = Array (Unlifted.map f arr)
+  fmap f arr = map (forget f) arr
```

Comment:
This is ugly, but the implementation is correct and I don't know of a way around this. It's a tricky part of HOFs, where we technically should use the implementation of `Data.fmap` to implement `map` since it is less restrictive and has the more specific implantation but we can't.

Reply:
Yes, `map` is less restrictive than `fmap`. As the implementation here demonstrates.